Add plugin: Move Files

nitishkhurana opened the pull request:
# I am submitting a new Community Plugin

## Repo URL
Link to my plugin: nitishkhurana/obsidian-move-files-plugin

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Zachatoo commented:
Remove the `styles.css` file from your release and your repo, since it's empty it's just unnecessary bloat that Obsidian would download.

["authorUrl": "",](https://github.com/nitishkhurana/obsidian-move-files-plugin/blob/b3007d993d32fc8506429ca0b86a8c583e50dc12/manifest.json#L8)
If you don't have a website to link to, please remove this line. Many people link to their GitHub profile page.

["fundingUrl": "",](https://github.com/nitishkhurana/obsidian-move-files-plugin/blob/b3007d993d32fc8506429ca0b86a8c583e50dc12/manifest.json#L9)
This is meant for links to services like _Buy me a coffee_, _GitHub sponsors_ and so on, if you don't have such a link remove this line.

[callback: () => {](https://github.com/nitishkhurana/obsidian-move-files-plugin/blob/b3007d993d32fc8506429ca0b86a8c583e50dc12/main.ts#L10)
This should be a [conditional command](https://docs.obsidian.md/Plugins/User+interface/Commands#Conditional+commands), with the condition being an active markdown file.

[await this.app.vault.copy(fileItem,targetPath);](https://github.com/nitishkhurana/obsidian-move-files-plugin/blob/b3007d993d32fc8506429ca0b86a8c583e50dc12/main.ts#L67-L68)
Prefer to use [`FileManager.renameFile()`](https://docs.obsidian.md/Reference/TypeScript+API/FileManager/renameFile) instead of `Vault.copy()` and `Vault.delete()` to also update all links to it depending on the user's preferences.

[new Notice(`Did not Move ${fileItem.name} to ${targetPath} as file already exists`);](https://github.com/nitishkhurana/obsidian-move-files-plugin/blob/b3007d993d32fc8506429ca0b86a8c583e50dc12/main.ts#L87)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[const imageRegex = /!\[\[(.*?)\]\]|!\[(.*?)\]\((.*?)(?:\|.*?)?\)/g;](https://github.com/nitishkhurana/obsidian-move-files-plugin/blob/b3007d993d32fc8506429ca0b86a8c583e50dc12/main.ts#L30)
This is going to capture all embeds, not just image embeds, see <https://help.obsidian.md/Linking+notes+and+files/Embed+files>. Please get image embeds from [`MetadataCache.getFileCache`](https://docs.obsidian.md/Reference/TypeScript+API/MetadataCache/getFileCache) instead.
```ts
const embeds = this.app.metadataCache.getFileCache(file)?.embeds ?? [];
const images = embeds
    .map(embed => this.app.metadataCache.getFirstLinkpathDest(embed.link, file))
    .filter(file => file.extension !== 'md');
```

nitishkhurana commented:
Addressed the review comments.
For the last review comment: As this plugin is for all the embedded files and not only for images, so only updated the variable name.

nitishkhurana commented:
Incorporated the further review comments.
